msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -525,22 +618,47 @@ public boolean removeAccount(@Nullable final IAccount account) {
                     "Requisite IAccount or IAccount fields were null. Insufficient criteria to remove IAccount."
             );
 
-            return false;
+            callback.onAccountsRemoved(false);
         }
 
         // FEATURE SWITCH: Set to false to allow deleting Accounts in a tenant-specific way.
         final boolean deleteAccountsInAllTenants = true;
 
         final String realm = deleteAccountsInAllTenants ? null : getRealm(account);
 
-        return !mPublicClientConfiguration
+        final boolean localRemoveAccountSuccess = !mPublicClientConfiguration
                 .getOAuth2TokenCache()
                 .removeAccount(
                         account.getEnvironment(),
                         mPublicClientConfiguration.getClientId(),
                         account.getHomeAccountIdentifier().getIdentifier(),
                         realm
                 ).isEmpty();
+
+        if (MSALControllerFactory
+                .brokerEligible(mPublicClientConfiguration.getAppContext(),
+                mPublicClientConfiguration.getDefaultAuthority(),
+                mPublicClientConfiguration)) {
+            //Remove the account from Broker
+            sBackgroundExecutor.submit(new Runnable() {
```

Comment:
move the Background operation logic to `BrokerMsalController().removeBrokerAccount` as discussed